extra/src/main/java/aima/extra/probability/ProbTensor.java

```diff
@@ -0,0 +1,153 @@
+package aima.extra.probability;
+
+import java.math.BigInteger;
+import java.math.MathContext;
+
+/**
+ * ProbTensor is a helper class which allows for ProbabilityNumber operations to
+ * return ProbabilityNumber objects that may contain invalid values. It is
+ * possible for intermediate calculations to result in invalid probability
+ * values, whereas the outcome after applying all the operations results in a
+ * valid probability value.
+ * 
+ * @author Nagaraj Poti
+ */
+public class ProbTensor {
+
+	public MathContext ARBITRARY_PRECISION = null;
+	
+	/**
+	 * Add two ProbabilityNumber operands and return the result.
+	 * 
+	 * @param operand1
+	 * @param operand2
+	 * 
+	 * @return ProbabilityNumber
+	 */
+	public ProbabilityNumber add(ProbabilityNumber operand1, ProbabilityNumber operand2) {
+		synchronized (ProbabilityNumber.class) {
```

Comment:
I recommend adding a class instance attribute that determines if checks are to be performed and extend the ProbabilityNumber API to accept an interval check flag as well as the existing math context flag.